clients/da-vinci-client/src/test/java/com/linkedin/davinci/storage/StorageServiceTest.java

```diff
@@ -121,4 +136,76 @@ public void testGetStoreAndUserPartitionsMapping() {
     expectedMapping.put(resourceName, partitionSet);
     Assert.assertEquals(storageService.getStoreAndUserPartitionsMapping(), expectedMapping);
   }
+
+  @Test
+  public void testCheckWhetherStoragePartitionsShouldBeKeptOrNot() throws NoSuchFieldException, IllegalAccessException {
+    StorageService mockStorageService = mock(StorageService.class);
+    SafeHelixManager manager = mock(SafeHelixManager.class);
+    StorageEngineRepository mockStorageEngineRepository = mock(StorageEngineRepository.class);
+    AbstractStorageEngine abstractStorageEngine = mock(AbstractStorageEngine.class);
+    mockStorageEngineRepository.addLocalStorageEngine(abstractStorageEngine);
+
+    String resourceName = "test_store_v1";
+    String storeName = "test_store";
+
+    when(abstractStorageEngine.getStoreVersionName()).thenReturn(resourceName);
+    abstractStorageEngine.addStoragePartition(0);
+    abstractStorageEngine.addStoragePartition(1);
+
+    String clusterName = "test_cluster";
+    VeniceConfigLoader mockVeniceConfigLoader = mock(VeniceConfigLoader.class);
+    VeniceServerConfig mockServerConfig = mock(VeniceServerConfig.class);
+    VeniceClusterConfig mockClusterConfig = mock(VeniceClusterConfig.class);
+    when(mockServerConfig.getDataBasePath()).thenReturn("/tmp");
+    when(mockVeniceConfigLoader.getVeniceServerConfig()).thenReturn(mockServerConfig);
+    when(mockVeniceConfigLoader.getVeniceClusterConfig()).thenReturn(mockClusterConfig);
+    when(mockVeniceConfigLoader.getVeniceClusterConfig().getClusterName()).thenReturn(clusterName);
+
+    List<AbstractStorageEngine> localStorageEngines = new ArrayList<>();
+    localStorageEngines.add(abstractStorageEngine);
+
+    SafeHelixDataAccessor helixDataAccessor = mock(SafeHelixDataAccessor.class);
+    when(manager.getHelixDataAccessor()).thenReturn(helixDataAccessor);
+    IdealState idealState = mock(IdealState.class);
+    when(helixDataAccessor.getProperty((PropertyKey) any())).thenReturn(idealState);
+    ZNRecord record = new ZNRecord("testId");
+    Map<String, Map<String, String>> mapFields = new HashMap<>();
+    Map<String, String> testPartitionZero = new HashMap<>();
+    Map<String, String> testPartitionOne = new HashMap<>();
+    testPartitionZero.put("lor1-app56585.prod.linkedin.com_1690", "LEADER");
+    testPartitionZero.put("lor1-app56614.prod.linkedin.com_1690", "STANDBY");
+    testPartitionZero.put("lor1-app110448.prod.linkedin.com_1690", "STANDBY");
+    testPartitionOne.put("lor1-app56586.prod.linkedin.com_1690", "LEADER");
+    testPartitionOne.put("lor1-app71895.prod.linkedin.com_1690", "STANDBY");
+    testPartitionOne.put("lor1-app111181.prod.linkedin.com_1690", "STANDBY");
```

Comment:
Instead of using actual LinkedIn host names, can you create general ones?

Reply:
Updated